### PR TITLE
Feature/ab2d 403 test containers

### PR DESCRIPTION
**Pull request by adaykin**
Adding in test containers

Add in a test container that gets built prior to the test running and has the same version of postgres running that is running in prod. The tests will use the container and testcontainers annotations to utilize the container. Upgrading some parts to junit 5 (jupiter)

### Checklist

- [x] Tests and linting pass
- [x] Code checked for PHI/PII exposure

### Security
What secure items does this touch, how does this PR affect our security posture, etc?

NA

### CODE REVIEW

File: common/src/test/java/gov/cms/ab2d/common/util/AB2DPostgresqlContainer.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public void start() {
+        super.start();
+        System.setProperty("DB_URL", container.getJdbcUrl());
+        System.setProperty("DB_USERNAME", container.getUsername());
+        System.setProperty("DB_PASSWORD", container.getPassword());
+    }
+
+    @Override
+    public void stop() {
+        //do nothing, JVM handles shut down
```
Comment (dkrylovsb) [inline common/src/test/java/gov/cms/ab2d/common/util/AB2DPostgresqlContainer.java]: We should set `container` to null here.

Comment (dkrylovsb) [inline common/src/test/java/gov/cms/ab2d/common/util/AB2DPostgresqlContainer.java]: We should call `super.stop()` I think.

File: common/src/test/java/gov/cms/ab2d/common/util/AB2DPostgresqlContainer.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private AB2DPostgresqlContainer() {
+        super(IMAGE_VERSION);
+    }
+
+    public static AB2DPostgresqlContainer getInstance() {
+        if (container == null) {
+            container = new AB2DPostgresqlContainer();
+        }
+        return container;
+    }
```
Comment (dkrylovsb) [inline common/src/test/java/gov/cms/ab2d/common/util/AB2DPostgresqlContainer.java]: Do we need to reuse the container across tests? If we use `new AB2DPostgresqlContainer()` instead in the tests, we will make sure each test starts with a clean DB.

File: worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java [hunk 3]
```diff
@@ -34,7 +34,7 @@
 import static net.logstash.logback.argument.StructuredArguments.keyValue;
 
 @Slf4j
-@Service
+@Service()
```
Comment (dkrylovsb) [inline worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java]: What's this?

Comment (adaykin) [reply worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java]: I think I was playing around with that when I was trying to get the worker module to work with JUnit5, I can revert

**Pull request merged**